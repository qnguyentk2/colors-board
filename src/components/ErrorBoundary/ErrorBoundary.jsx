import React from 'react';
import PropTypes from 'prop-types';
// import { captureReactError } from 'common/sentry';

export default class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      error: false,
    };
  }

  componentDidCatch() {
    this.setState({ error: true });
    // captureReactError(error, errorInfo);
  }

  render() {
    if (this.state.error) {
      return (
        <div
          image="/icon-error-SPlatform.gif"
          description={(
            <span>
              We&aposre sorry â€” something&aposs gone wrong.
              <br />
              Our team has been notified.
            </span>
          )}
        />
      );
    }
    return this.props.children;
  }
}

// eslint-disable-line
ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
};
