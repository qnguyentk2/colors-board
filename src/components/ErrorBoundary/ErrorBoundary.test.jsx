/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
// import { MemoryRouter } from 'react-router';
import toJSON from 'enzyme-to-json';
import ErrorBoundary from './ErrorBoundary';

const propsData = {
  children: {}
}

const setup = props => {
  const wrapper = shallow(
    <ErrorBoundary {...props} />
  );

  return { wrapper };
}

describe('Test PageLayout', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup(propsData);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
})