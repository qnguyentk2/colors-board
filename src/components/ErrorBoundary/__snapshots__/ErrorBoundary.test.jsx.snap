// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PageLayout renders without crashing 1`] = `<Component />`;
